src/lightly_train/_commands/train_task.py

```diff
@@ -230,6 +230,20 @@ def _train_task_from_config(config: TrainTaskConfig) -> None:
     logger.info(f"Args: {helpers.pretty_format_args(args=initial_config)}")
     logger.info(f"Using output directory: '{out_dir}")
 
+    # Load checkpoint context if resuming or further fine-tuning.
+    checkpoint_ctx = helpers.BaseCheckpointContext.from_config(
+        fabric=fabric, config=config, out_dir=out_dir
+    )
+    if checkpoint_ctx:
+        model_init_args = checkpoint_ctx.metadata.model_init_args
+        if (saved_model_name := model_init_args.get("model_name")) != config.model:
+            raise ValueError(
+                f"The model name in the checkpoint or model weights file('{saved_model_name}') "
+                f"does not match the provided model name ('{config.model}')."
+            )
+    else:
+        model_init_args = {}
```

Comment:
Move the logic as far down in the function as possible, right before `model_init_args` is used the first time. There is code that is more important that should be run before just below (e.g. logging system information).